microsoft/PowerToys
# [ColorPicker] Ensure that the hotkey always works

## Summary of the Pull Request

If you press the ColorPicker hotkey while it's enable and not running for any reason, nothing will happen. Also, you could see multiple instances of ColorPicker running happily one beside another. This PR should address both problems.

**What is this about:**

For more info, see #8831 #10525

**What is include in the PR:** 

* Replaced `Application.Current.Shutdown()` with `Environment.Exit(0)` to close the app if the mutex can't be captured.
* Now using a local mutex to ensure you can run multiple instances in different sessions.
* Use the hotkey API in the module interface to invoke the app.

**How does someone test / validate:** 

* Try running ColorPickerUI.exe manually several times to ensure only one instance is running.
* Try enabling ColorPicker, killing it in the Task manager, and then pressing the hotkey. It should boot up and activate.
* Reproduce circumstances of linked issues.

## Quality Checklist

- [x] **Linked issue:**  #8831 #10525

## Contributor License Agreement (CLA)
A CLA must be signed. If not, go over [here](https://cla.opensource.microsoft.com/microsoft/PowerToys) and sign the CLA.
